### PR TITLE
Move duplicated logic to scripts

### PR DESCRIPTION
**Pull request by yuvalyaron**
# Improves #3299

## What is being addressed

Moved duplicated logic to shared scripts

### CODE REVIEW

File: devops/scripts/construct_tre_url.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+set -o errexit
+set -o pipefail
+set -o nounset
```
Comment (tamirkamara) [inline devops/scripts/construct_tre_url.sh:4]: Should these be here if the script is sourced?

File: devops/scripts/convert_azure_env_to_arm_env.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+set -o errexit
+set -o pipefail
+set -o nounset
```
Comment (tamirkamara) [inline devops/scripts/convert_azure_env_to_arm_env.sh:4]: Should these be here if the script is sourced?

**Review by tamirkamara — APPROVED** [review]

File: devops/scripts/load_and_validate_env.sh [hunk 3]
```diff
@@ -43,11 +48,17 @@ else
     # shellcheck disable=SC2046
     export $(yq e "$GET_LEAF_KEYS|$TF_KEYS| $FORMAT_FOR_ENV_EXPORT" config.yaml)
 
+    # Source AZURE_ENVIRONMENT and setup the  ARM_ENVIRONMENT based on it
```
[the remark this answers is not in the document]
Comment (yuvalyaron) [reply devops/scripts/load_and_validate_env.sh:51]: this is part of @LizaShak's PR which wasn't merged yet

File: devops/scripts/load_and_validate_env.sh [hunk 4]
```diff
@@ -43,11 +48,17 @@ else
...
     AZURE_ENVIRONMENT=$(az cloud show --query name --output tsv)
     export AZURE_ENVIRONMENT
 
-    declare -A cloudapp_endpoint_suffix=( ["public"]="cloudapp.azure.com" ["usgovernment"]="cloudapp.usgovcloudapi.net")
-    export TRE_URL=${TRE_URL:-https://${TRE_ID}.${LOCATION}.${cloudapp_endpoint_suffix[${ARM_ENVIRONMENT}]}}
+    # The ARM Environment is required by terrafform to indicate the destination cloud.
```
[the remark this answers is not in the document]
Comment (yuvalyaron) [reply devops/scripts/load_and_validate_env.sh:55]: this is part of @LizaShak's PR which wasn't merged yet

**Comment by azuretresussex** [comment]
Hello, I'm deploying an AzureTRE using deploy_tre.yml, and CICD currently failing with the following output.

Prepare all required actions
Getting action download info
Run ./.github/actions/devcontainer_run_command
Run azure_env="AzureCloud"
Run source devops/scripts/construct_tre_url.sh
/home/runner/work/_temp/15921fb8-6b7f-497f-88ed-b3e0210eea03.sh: line 1: devops/scripts/construct_tre_url.sh: No such file or directory
Error: Process completed with exit code 1.

Could this be related to the change described in #3371 ?

**Comment by marrobi** [comment]
Hello,  @azuretresussex , could you create a new issue for this  query - https://github.com/microsoft/AzureTRE/issues ?

Also when you create the issue please confirm if you are using the AzureTRE-Deployment repository or the AzureTRE repository to deploy? 

Thanks.

**Pull request merged**